frontend/src/components/modal-compare/modal.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, Dispatch } from 'react';
+
+import fillCompare from '@assets/images/icon_balance.svg';
+
+import styles from './styles.module.scss';
+
+interface ModalProps {
+  carName: string;
+  carDescription: string;
+  isHidden: boolean;
+  setIsHidden: Dispatch<boolean>;
+}
+
+const Modal: FC<ModalProps> = ({
+  carName,
+  carDescription,
+  isHidden,
+  setIsHidden,
+}) => {
+  const closeModal = (): void => {
+    setIsHidden(!isHidden);
+  };
+
+  return (
+    <div className={styles.container} hidden={isHidden}>
+      <div className={styles.content}>
+        <button className={styles.close} onClick={(): void => closeModal()}>
+          ×
+        </button>
```

Comment:
This `onClick` handler can be simplified. If you pass the anon func to the `onClick`, you can directly call the `setIsHidden`. But if you already have the `closeModal` defined, you can pass it directly to the `onClick`

```suggestion
        <button className={styles.close} onClick={closeModal}>
          ×
        </button>
```